File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxprofit=0;
        int buy=prices[0];
        for(int i=1;i<prices.length;i++){
            maxprofit=Math.max(maxprofit,prices[i]-buy);
            buy=Math.min(prices[i],buy);
        }
        return maxprofit;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_price=prices[0]
        max_pro=0
        for i in range(len(prices)):
            max_pro=max(max_pro,prices[i]-min_price)
            min_price=min(min_price,prices[i])
        return max_pro